opm/material/densead/Evaluation.hpp

```diff
@@ -168,15 +186,15 @@ class Evaluation
     {
         // while the values are multiplied, the derivatives follow the product rule,
         // i.e., (u*v)' = (v'u + u'v).
-        const ValueType& u = this->value_;
-        const ValueType& v = other.value_;
-        for (unsigned varIdx = 0; varIdx < size; ++varIdx) {
-            const ValueType& uPrime = this->derivatives_[varIdx];
-            const ValueType& vPrime = other.derivatives_[varIdx];
+        ValueType& u = data_[ valuepos_ ];
```

Comment:
u is used a bit below. So either const ValueType& u and then data_[ valuepos_ ] below or the way I have put it. I'm usually in favour of keeping the places where valuepos_ is used to a minimum.